pkg/cvo/status.go

```diff
@@ -128,26 +132,25 @@ func mergeOperatorHistory(config *configv1.ClusterVersion, desired configv1.Rele
 	}
 
 	// TODO: prune Z versions over transitions to Y versions, keep initial installed version
-	pruneStatusHistory(config, 50)
+	pruneStatusHistory(config, MaxHistory)
 
 	config.Status.Desired = desired
 }
 
+// pruneStatusHistory maintains history size at MaxHistory by removing entry at index MaxHistory
+// unless that entry is a completed update in which case entry at MaxHistory-1 is removed thereby
+// retaining the initial completed version.
 func pruneStatusHistory(config *configv1.ClusterVersion, maxHistory int) {
 	if len(config.Status.History) <= maxHistory {
 		return
 	}
-	for i, item := range config.Status.History {
-		if item.State != configv1.CompletedUpdate {
-			continue
-		}
-		// guarantee the last position in the history is always a completed item
-		if i >= maxHistory {
-			config.Status.History[maxHistory-1] = item
-		}
-		break
+	if config.Status.History[maxHistory].State == configv1.CompletedUpdate {
+		item := config.Status.History[maxHistory]
```

Comment:
I doubt we will drop `MaxHistory` below the current 50, but if we do, is it worth teaching this to prefer the oldest `>=maxHistory` index that's completed?

Reply:
Currently, because `pruneStatusHistory` gets called each time a new entry is added the history index will never exceed `Maxhistory`. This could change in the future but most likely if it does it will be part of a more robust history pruning re-do, e.g. implementing our enhancement, which will change things significantly anyway.